### PR TITLE
Extract All Custom Swagger Vars To Swagger Options

### PR DESCRIPTION
**Pull request by DenisPopov15**

### CODE REVIEW

**Comment by DenisPopov15** [comment]
@alexkravets updated

File: lib/swagger/generator.rb [hunk 1]
```diff
@@ -276,22 +292,23 @@ def generate_swagger_paths(base_path=false, hide_custom_paths=false)
...
       custom_routes.each do |route|
-        path = route[:path].gsub(base_path, '')
+        path = route[:path].gsub(swagger_base_path, '')
         if path.include?(':id')
           path = path.gsub(':id', '{id}')
           swagger_path "#{ path }" do
             operation route[:method] do
-              key :tags, [ plural ]
+              key :tags, tags
               key :summary, 'Custom method, Please Overwrite this Method Manually'
```
Comment (alexkravets) [inline lib/swagger/generator.rb]: This description is not clear. Use: 'Default action schema is used, custom Swagger description is required.'

File: lib/swagger/generator.rb [hunk 2]
```diff
@@ -276,22 +292,23 @@ def generate_swagger_paths(base_path=false, hide_custom_paths=false)
...
               key :description, 'Dont describe reqired fields and response'
```
Comment (alexkravets) [inline lib/swagger/generator.rb]: This description doesn't make sense. Please re-phrase and check grammar.

File: lib/swagger/generator.rb [hunk 3]
```diff
@@ -18,6 +26,18 @@ module SwaggerGenerator
     # http://files.slatestudio.com/gG30
     ALLOW_TYPES  = %w(Object String Integer Array Date Mongoid::Boolean Symbol)
 
+    def swagger_options(options)
```
Comment (alexkravets) [inline lib/swagger/generator.rb:26]: This method doesn't look good, let's put default values back to the top of `generate_swagger_schemas` and `generate_swagger_paths`.

So in the `swagger_options` you'll have:

`swagger_base_path = options[:base_path]`

And on the top of `generate_swagger_paths`:

`swagger_base_path ||= ''`

Comment (alexkravets) [inline lib/swagger/generator.rb:26]: So the way this works is:
1. set custom values with `swagger_options`
2. if value is set it's used as is
3. if value is missing default value is set

**Comment by DenisPopov15** [comment]
@alexkravets updated

File: lib/swagger/generator.rb [hunk 4]
```diff
@@ -18,6 +23,14 @@ module SwaggerGenerator
...
+      self.swagger_base_path           = options[:base_path]
+      self.swagger_collection_name     = options[:collection_name]
+      self.swagger_relative_path       = options[:relative_path]
+      self.ignore_custom_actions       = options[:ignore_custom_actions]
```
Comment (alexkravets) [inline lib/swagger/generator.rb]: I think this also should have `swagger_` prefix that illustrates that this class attribute belongs to to swagger module.

File: lib/swagger/generator.rb [hunk 5]
```diff
@@ -68,24 +81,20 @@ def fetch_required_fields(resource_class)
...
-    def generate_swagger_schemas(name=false)
-      name ||= resource_name
-
-      if resource_class
-        # TODO: add support for resource class option
-      else
-        resource_class = name.constantize
-      end
+    def generate_swagger_schemas
+      self.swagger_resource_class_name ||= resource_name
```
Comment (alexkravets) [inline lib/swagger/generator.rb:85]: Why do you need `self.` here?

**Comment by alexkravets** [comment]
Double check that you really need `self.`, fix `swagger_` prefix and good to go.

**Comment by DenisPopov15** [comment]
@alexkravets OK. `self.` required, double checked

**Comment by DenisPopov15** [comment]
@alexkravets Can I merge?

**Comment by alexkravets** [comment]
yes

**Pull request merged**